lib/document.js

```diff
@@ -213,7 +213,7 @@ for (const i in EventEmitter.prototype) {
  * @instance
  */
 
-Document.prototype.schema;
+Document.prototype.$__schema;
```

Comment:
We should keep `Document.prototype.schema` for backwards compatibility. `$__schema` is what Mongoose will use internally, but there will still be a `schema` property that developers can use if they need to. Also, `$__schema` should be marked with `@api private` so it isn't part of the public API.